servicetalk-tcp-netty-internal/src/main/java/io/servicetalk/tcp/netty/internal/TcpServerBinder.java

```diff
@@ -169,6 +175,84 @@ protected void handleSubscribe(Subscriber<? super ServerContext> subscriber) {
         };
     }
 
+    /**
+     * Wraps the connection function with early and late acceptors.
+     *
+     * @param connection the connection function which establishes the pipeline.
+     * @param channel the netty channel on which the connection is run.
+     * @param executionContext The {@link ExecutionContext} to use for the bind socket.
+     * @param earlyConnectionAcceptor the {@link EarlyConnectionAcceptor} to filter newly accepted sockets early.
+     * @param lateConnectionAcceptor the {@link LateConnectionAcceptor} to filter newly accepted sockets.
+     * @param connectionAcceptor The {@link ConnectionAcceptor} used to filter newly accepted sockets.
+     * @param <CC> The type of {@link ConnectionContext} that is created for each accepted socket.
+     * @return the wrapped connection function with the connection acceptors (if any).
+     */
+    private static <CC extends ConnectionContext> Single<CC> wrapConnectionAcceptors(
+            Single<CC> connection,
+            final Channel channel,
+            final ExecutionContext<?> executionContext,
+            @Nullable final EarlyConnectionAcceptor earlyConnectionAcceptor,
+            @Nullable final LateConnectionAcceptor lateConnectionAcceptor,
+            @Nullable final InfluencerConnectionAcceptor connectionAcceptor) {
+        final Executor offloadExecutor = executionContext.executor();
+
+        if (earlyConnectionAcceptor != null) {
+            ReducedConnectionInfo info = new ReducedConnectionInfo() {
+                @Override
+                public SocketAddress localAddress() {
+                    return channel.localAddress();
+                }
+
+                @Override
+                public SocketAddress remoteAddress() {
+                    return channel.remoteAddress();
+                }
+
+                @Override
+                public ExecutionContext<?> executionContext() {
+                    return executionContext;
+                }
+            };
+
+            final boolean ioThreadSupported = executionContext.ioExecutor().isIoThreadSupported();
+            final NettyIoExecutor ioExecutor = fromNettyEventLoop(channel.eventLoop(), ioThreadSupported);
```

Comment:
Looks like we can move `ExecutionContextUtils.fromChannel` to transport-netty-internal in order to reuse here, but won't be able to move `channelExecutionContext` due to dependency on `HttpExecutionContext`. It's ok, we can have a similar utility that uses `DelegatingExecutionContext`.